core/src/test/scalanative/sttp/client3/testing/SyncHttpTestExtensions.scala

```diff
@@ -82,12 +82,26 @@ trait SyncHttpTestExtensions {
 
   "download file" - {
 
-    "download a binary file using asFile" in {
-      withTemporaryNonExistentFile { file =>
-        val req = basicRequest.get(uri"$endpoint/download/binary").response(asFile(file))
-        val resp = req.send(backend)
-        md5FileHash(resp.body.right.get).map { _ shouldBe binaryFileMD5Hash }
-      }
+    // TODO this test do not verify actual downloaded file
```

Comment:
it checks the md5, so it does?

Reply:
so maybe my description is incorrect as our binary file from test was not saved properly while tests was passing

Reply:
hm that's weird - how did the md5 match then? ;) I guess we can keep both tests

Reply:
ah, maybe `resp.body.right.get` was `None`?

Reply:
I will check on current master give me some time

Reply:
Yes you are right - to a degree, resp.body.right.get is good but `md5FileHash(resp.body.right.get)` is empty

Reply:
In fact there is some strange error with md5Hash function -> it always return empty string

Reply:
https://github.com/scala-native/scala-native/pull/2175